libs/libc/dirent/lib_telldir.c

```diff
@@ -61,15 +56,11 @@
 
 off_t telldir(FAR DIR *dirp)
 {
-  struct fs_dirent_s *idir = (struct fs_dirent_s *)dirp;
-
-  if (!idir || !idir->fd_root)
+  if (dirp)
```

Comment:
```suggestion
  if (dirp != NULL)
```

Reply:
Done!

Reply:
> What was the motivation for this change?
> 
> For a change of this size nature all the file system will need to tested and result posted as well. the statistics on the impact on memory ram/flash.

@davids5 
Statistics for three situation: nodir, dir, dir_with_fchdir_dirfd
```
djz:nuttx$ ls -l vela_cp.bin 
-rwxrwxr-x 1 djz djz 2066272 8月   9 10:36 vela_cp.bin

djz:nuttx$ ls -l vela_cp_dir.bin 
-rwxrwxr-x 1 djz djz 2066260 8月   9 10:56 vela_cp_dir.bin

djz:nuttx$ ls -l vela_cp_dir_with_fchdir_dirfd.bin 
-rwxrwxr-x 1 djz djz 2066428 8月   9 11:10 vela_cp_dir_with_fchdir_dirfd.bin

no_dir:
cp> free
                   total       used       free    largest  nused  nfree
        Umem:     519096     210392     308704     304384    672      8
dir:
cp> free
                   total       used       free    largest  nused  nfree
        Umem:     519032     210392     308640     304864    672     10
dir_with_fchdir_dirfd:
cp> free
                   total       used       free    largest  nused  nfree
        Umem:     519096     210392     308704     304928    672      8
        
filesystem usage：
no_dir:
cp> df -h
  Filesystem    Size      Used  Available Mounted on
  rpmsgfs      2298M      121M      2177M /data
  romfs         512B      512B         0B /etc
  littlefs      256K        8K       248K /misc
  procfs          0B        0B         0B /proc
  tmpfs         512B      512B         0B /tmp
  rpmsgfs      1718K     1718K         0B /vendor

dir:
cp> df -h
  Filesystem    Size      Used  Available Mounted on
  rpmsgfs      2298M      121M      2177M /data
  romfs         512B      512B         0B /etc
  littlefs      256K        8K       248K /misc
  procfs          0B        0B         0B /proc
  tmpfs         512B      512B         0B /tmp
  rpmsgfs      1718K     1718K         0B /vendor

dir_with_fchdir_dirfd:
cp> df -h
  Filesystem    Size      Used  Available Mounted on
  rpmsgfs      2298M      122M      2176M /data
  romfs         512B      512B         0B /etc
  littlefs      256K        8K       248K /misc
  procfs          0B        0B         0B /proc
  tmpfs         512B      512B         0B /tmp
  rpmsgfs      1718K     1718K         0B /vendor
```